module3/homework/advanced_calculator/advancedCalculator.cpp

```diff
@@ -0,0 +1,132 @@
+#include "advancedCalculator.hpp"
+#include <functional>
+#include <iostream>
+#include <map>
+#include <math.h>
+#include <regex>
+#include <sstream>
+#include <stdexcept>
+#include <stdlib.h>
+
+CalculationException::CalculationException(std::string p_msg, ErrorCode errorCode_)
+    : logic_error(p_msg), errorCode(errorCode_)
+{
+}
+
+const char* CalculationException::what() const throw()
+{
+    std::ostringstream o;
+    o << logic_error::what();
+    return o.str().c_str();
+}
+
+ErrorCode parse(const std::string& input, double& a, double& b, std::string& oper)
+{
+
+    if (input.find_first_not_of("0123456789-+*$/^%!., ") != std::string::npos)
```

Comment:
Maybe make this string (inside find_first_not_of function) as const variable?

Reply:
+1